Ci improve if doc cpp

nicoloboschi opened the pull request.

nicoloboschi committed:
[ci] Reduce runners load for pulls that affects only doc or cpp changes

lhotari commented on .github/changes-filter.yaml:
```diff
@@ -8,6 +8,7 @@ docs:
   - '.asf.yaml'
   - '*.md'
   - '**/*.md'
+  - '**/*.yaml'
```
this rule would also match `.github/workflows/*.yaml` is that intentional? the problem is that workflow changes wouldn't be tested.

nicoloboschi replied:
it's just temporary to verify my pull

nicoloboschi committed:
[ci] Temporarily remove required jobs to unblock CI (#17539)

* Following the action plan defined in https://lists.apache.org/thread/yh72bmlf2w97593c5d5pqf3jz1ldb501 .
* A lazy consensus was reached on this matter to fix CI following the action plan.

cbornet committed:
[PIP-193] Support Transform Function with LocalRunner (#17445)

nicoloboschi committed:
[ci] Disable master branch protection rules (#17547)

- Follow up on #17539 changes due to error message https://lists.apache.org/thread/hj8b9n1n29vg4c6oslz6n1j230f9v284

- Following the action plan defined in https://lists.apache.org/thread/yh72bmlf2w97593c5d5pqf3jz1ldb501 .
- A lazy consensus was reached on this matter to fix CI following the action plan.

lhotari committed:
[improve][CI] Remove "Cancel duplicate workflows" workflow (#17490)

- there's no need for this anymore since GitHub Actions has built-in support
  for cancelling duplicate workflows and this is used in Pulsar CI workflows.

lhotari committed:
[improve][CI] Optimize "Pulsar Bot" workflow (#17491)

- move the filter from the build step to the build job
   - only start the build job if the comment includes "/pulsarbot"
- no need to checkout code
- no need to run the tune VM action

nicoloboschi committed:
[ci] Extract FLAKY tests suite to a dedicated workflow (#17542)

Co-authored-by: Lari Hotari <lhotari@users.noreply.github.com>

tisonkun committed:
Delete project-flaky-test.yml (#17529)

equanz committed:
[feat][cpp] Add async receive function to C API (#17452)

nicoloboschi committed:
add final jobs to mark required workflows as completed even if they are skipped

nicoloboschi committed:
better impl

nicoloboschi committed:
test parametrized job id

nicoloboschi committed:
move non required system tests to flaky job

nicoloboschi committed:
fix flaky wf

nicoloboschi committed:
fix wf

Technoboy- committed:
Use `safeRun` to log thread exception. (#17484)

Technoboy- committed:
[improve][admin] Make `getBundleRange` async (#17402)

nicoloboschi committed:
move back system-tests to main workflow

nicoloboschi committed:
fix latest step

nicoloboschi committed:
add macos build
